### PR TITLE
zh-trans: add translate-compose-kubernetes.md

### PR DESCRIPTION
**Pull request by AdamDang**
zh-trans: add translate-compose-kubernetes.md

### CODE REVIEW

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,849 @@
...
+
+{{% capture overview %}}
+
+<!--
+What's Kompose? It's a conversion tool for all things compose (namely Docker Compose) to container orchestrators (Kubernetes or OpenShift).
+
+More information can be found on the Kompose website at [http://kompose.io](http://kompose.io).
+-->
+
+Kompose 是什么？它是个转换工具，可用于由容器编排器（Kubernetes 或 OpenShift）组成（即 Docker Compose）的所有内容。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]:
```suggestion
Kompose 是什么？它是个转换工具，可将 compose（即 Docker Compose）所组装的所有内容转换成容器编排器（Kubernetes 或 OpenShift）可识别的形式。
```

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,849 @@
...
+We have multiple ways to install Kompose. Our preferred method is downloading the binary from the latest GitHub release.
+
+## GitHub release
+
+Kompose is released via GitHub on a three-week cycle, you can see all current releases on the [GitHub release page](https://github.com/kubernetes/kompose/releases).
+-->
+
+## 安装 Kompose
+
+我们有很多种方式安装 Kompose。首选方式是从最新的 GitHub 发布版本下载二进制文件。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: 发布页面

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,849 @@
...
+- 文档
+  - [构建和推送 Docker 镜像](#build-and-push-docker-images)
+  - [替代转换](#alternative-conversions)
+  - [标签](#labels)
+  - [重启](#restart)
+  - [Docker Compose 版本](#docker-compose-versions)
+
+
+Kompose has support for two providers: OpenShift and Kubernetes.
+You can choose a targeted provider using global option `--provider`. If no provider is specified, Kubernetes is set by default.
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: 以上两行漏翻

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,849 @@
...
+INFO OpenShift file "result-deploymentconfig.yaml" created  
+INFO OpenShift file "result-imagestream.yaml" created  
+```
+
+<!--
+It also supports creating buildconfig for build directive in a service. By default, it uses the remote repo for the current git branch as the source repo, and the current branch as the source branch for the build. You can specify a different source repo and branch using ``--build-repo`` and ``--build-branch`` options respectively.
+-->
+
+kompose 还支持为服务中的构建指令创建 buildconfig。
+默认情况下，它使用当前 git 分支的远程 repo 作为源 repo，使用当前分支作为构建的源分支。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]:
```suggestion
默认情况下，它使用当前 git 分支的 remote 仓库作为源仓库，使用当前分支作为构建的源分支。
```

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,849 @@
...
+您可以分别使用 ``--build-repo`` 和 ``--build-branch`` 选项指定不同的源 repo 和分支。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]:
```suggestion
您可以分别使用 ``--build-repo`` 和 ``--build-branch`` 选项指定不同的源仓库和分支。
```

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,849 @@
...
+INFO Successfully created Deployment: foo         
+
+Your application has been deployed to Kubernetes. You can run 'kubectl get deployment,svc,pods,pvc' for details.
+```
+
+<!--
+In order to disable the functionality, or choose to use BuildConfig generation (with OpenShift) `--build (local|build-config|none)` can be passed.
+-->
+
+要想禁用该功能，或者使用 BuildConfig 生成（在 OpenShift 中），可以通过传递 `--build (local|build-config|none)` 参数来实现。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: generation 是不是 BuildConfig 中的版本，也就是“代次”？

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,849 @@
...
+- CLI
+  - [`kompose convert`](#kompose-convert)
+  - [`kompose up`](#kompose-up)
+  - [`kompose down`](#kompose-down)
+- 文档
+  - [构建和推送 Docker 镜像](#build-and-push-docker-images)
+  - [替代转换](#alternative-conversions)
+  - [标签](#labels)
+  - [重启](#restart)
+  - [Docker Compose 版本](#docker-compose-versions)
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: 注意这里的链接，链接目标都是本文中的标题。标题被翻译后，链接会失效。
所以，所有翻译后的标题都要有`{}` 形式的显式ankor。例如，对于其他转换方式：

```
- [其他转换方式](#alternative-conversions)


## 其他转换方式 {#alternative-conversions}
```

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md (hunk 1 above)
Comment (AdamDang) [reply content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: Done!

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md (hunk 2 above)
Comment (AdamDang) [reply content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: Done！

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,849 @@
...
+|----------------------|-------------------------------------|
+| kompose.service.type | nodeport / clusterip / loadbalancer |
+| kompose.service.expose| true / hostname |
+
+
+{{< note >}}
+<!--
+The `kompose.service.type` label should be defined with `ports` only, otherwise `kompose` will fail.
+-->
+`kompose.service.type` 标签应该只用`端口`来定义，否则 `kompose` 会失败。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: 端口不要翻译

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,849 @@
...
+| `""`                       | controller object | `Always`            |
+| `always`                   | controller object | `Always`            |
+| `on-failure`               | Pod               | `OnFailure`         |
+| `no`                       | Pod               | `Never`             |
+-->
+
+| `docker-compose` `restart` | 创建的对象          | Pod `restartPolicy` |
+|----------------------------|-------------------|---------------------|
+| `""`                       | controller object | `Always`            |
+| `always`                   | controller object | `Always`            |
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: 以上两行的第二列，“控制器对象”

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md (hunk 3 above)
Comment (AdamDang) [reply content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: Done！

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md (hunk 4 above)
Comment (AdamDang) [reply content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: Done！

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md (hunk 5 above)
Comment (AdamDang) [reply content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: Done！

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md (hunk 7 above)
Comment (AdamDang) [reply content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: 学习了！Thanks！

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md (hunk 8 above)
Comment (AdamDang) [reply content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: done!

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md (hunk 9 above)
Comment (AdamDang) [reply content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: done！

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md (hunk 7 above)
Comment (AdamDang) [reply content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: 实验了一下，上面的方法不行，直接改成中文就可以了。
```sh
## 用户指南
- CLI
  - [`kompose convert`](#kompose-convert)
  - [`kompose up`](#kompose-up)
  - [`kompose down`](#kompose-down)
- 文档
  - [构建和推送 Docker 镜像](#构建和推送-Docker-镜像)
  - [其他转换方式](#其他转换方式)
  - [标签](#标签)
  - [重启](#重启)
  - [Docker Compose 版本](#Docker-Compose-版本)
```

**Comment by tengqm** [comment]
@AdamDang The preview of this work is at here: https://5c26f115ab4431000a07cc52--k8s-v1-12.netlify.com/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes/

Please do a self-check and fix other issues shown in the preview.

File: content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md (hunk 7 above)
Comment (tengqm) [reply content/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes.md]: Yes. I saw it. The method I showed you should work and it has been employed in some other places in website, but it is good to know that non-ASCII links are well supported!

Please continue check the output of this page in the preview pages. There seem quite a few places in need of more work.

**Comment by tengqm** [comment]
@AdamDang Each time you push a new commit, the netlify task will be triggered automatically thus generating a new preview version.

**Comment by AdamDang** [comment]
@tengqm  How about the preview url? It's changed with new commit or keep the same ?

**Comment by tengqm** [comment]
@AdamDang Not sure. You can check it before and after you pusing a commit. The comment left by the netlify bot changes when new commit is pushed, the hash code is always the latest one, e.g. d9dc870

**Comment by tengqm** [comment]
@AdamDang Follow the link provided by netlify bot which will lead you to the preview.

**Comment by AdamDang** [comment]
@tengqm the URL is changed to:
https://5c27437b6cf67a00089b0d5f--k8s-v1-12.netlify.com/zh/docs/tasks/configure-pod-container/translate-compose-kubernetes/

Everything looks OK now!
Thanks a lot!

**Comment by Hartmarken348** [comment]
Translate construct kubernates

**Pull request merged**